src/twisted/conch/ssh/keys.py

```diff
@@ -1241,8 +1396,20 @@ def verify(self, signature, data):
                 k = k.public_key()
             verifier = k.verifier(
                 signature, hashes.SHA1())
-        else:
-            raise BadKeyError("unknown key type %s" % (self.type(),))
+
+        elif keyType == 'EC':
```

Comment:
Ok, so now I see that we can't fall out the bottom of this because `type()` raises exceptions for unknown types. In that case, let's not bother with the `elif` here: there's no point in having the conditional because there can't be anything else. You should use `else` and `assert` the value instead. That applies to all of these cases.